### PR TITLE
Add typescript support, dotenv and inversion of control

### PR DESCRIPTION
**Pull request by jKelio**
solves #3 #8 #9

### CODE REVIEW

**Comment by jKelio** [comment]
@robertdue ready for review to solve issue #3

Merging with squash?

File: main.ts [hunk 1]
```diff
@@ -18,19 +18,19 @@ const handle = () => {
...
         } else {
-            robot.moveMouse(mouse.x - MOVE_MARGIN, mouse.y - MOVE_MARGIN);
+            moveMouse(mouse.x - MOVE_MARGIN, mouse.y - MOVE_MARGIN);
             direction = "down";
         }
         console.log("new mouse position: " + x + "|" + y)
     }
 
-    var mouse = robot.getMousePos();
+    mouse = getMousePos();
```
Comment (robertdue) [inline main.ts:30]: unnecessary to save this global

Comment (jKelio) [reply main.ts]: re-review already requested

**Comment by robertdue** [comment]
Imho Inversify is really unnecessary here. Since there's just a very small amount of classes its way more easy and better readable if we use plain typescript classes.

**Review by robertdue — CHANGES_REQUESTED** [review]
Imho Inversify is really unnecessary here. Since there's just a very small amount of classes its way more easy and better readable if we use plain typescript classes.

**Comment by jKelio** [comment]
We are already using plain typescript classes enhanced with the adantages of a lightweight and simple inversion of control.

Due to the official documentation of inversify it has been developed with 4 main goals:

1. Allow JavaScript developers to write code that adheres to the SOLID principles.
2. Facilitate and encourage the adherence to the best OOP and IoC practices.
3. Add as little runtime overhead as possible.
4. Provide a [state of the art development experience](https://github.com/inversify/InversifyJS/blob/master/wiki/ecosystem.md).

Why we shouldn't use the well-known inversion of control pattern?

**Review by robertdue — APPROVED** [review]

**Pull request merged**